src/securechip/securechip.c

```diff
@@ -309,20 +318,70 @@ bool securechip_setup(securechip_interface_functions_t* ifs)
             return false;
         }
     }
+    return true;
+}
+#endif
 
-    // Validate config.
-    if (is_config_locked && is_data_locked) {
-        bool same_config = false;
-        if (atcab_cmp_config_zone(_configuration, &same_config) != ATCA_SUCCESS) {
-            return false;
-        }
-        if (!same_config) {
-            return false;
-        }
+static bool _verify_config(void)
+{
+    bool is_config_locked = false;
+    ATCA_STATUS result = atcab_is_locked(LOCK_ZONE_CONFIG, &is_config_locked);
```

Comment:
This could be a `_zone_is_locked` function to be consistent with `_slot_is_locked`. But I don't feel super strongly about it.